lib/msprime.c

```diff
@@ -406,6 +406,13 @@ msp_alloc_segment(msp_t *self, double left, double right,
     return seg;
 }
 
+static void
+msp_free_recomb_map(msp_t *self)
+{
+    recomb_map_free(self->recomb_map);
```

Comment:
I think we can just have ``recomb_map_free(&self->recomb_map)`` in msp_free. Once the struct has been zero'd this is safe (and that's the first thing we do in msp_alloc).

Reply:
Is it expected that if `msp_alloc` fails then a client should call `msp_free`?